core/dbt/parser/schemas.py

```diff
@@ -887,9 +892,20 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
             if found_nodes:
                 # There might be multiple disabled nodes for this model
                 for node in found_nodes:
-                    # We're saving the patch_path because we need to schedule
-                    # re-application of the patch in partial parsing.
-                    node.patch_path = source_file.file_id
+                    # If this yaml file is enabled but the project config is not, we need to move
+                    # the node from disabled to manifest.nodes
+                    if patch.config.get("enabled"):
+                        test_from = {"key": block.target.yaml_key, "name": block.target.name}
```

Comment:
The "test_from" variation on add_node is only for tests, not regular nodes. I don't think we should need to do "add_node" plus "remove_node'. I would just do something like popping it from "disabled" and adding it to "nodes".  Also we can't really do this if there are multiple disabled nodes, since we wouldn't know which one to enable. So I think we'll have to limit this to cases where there's only one disabled node with this unique_id.  You also shouldn't need to do a ref_lookup to get the unique_id, the node will already have a unique_id in it. So there would be an if/else after "if patch.config.get("enabled") with len(found_nodes) == 1, plus throwing an error if len is more than 1. 

I think it might be possible to just add the node to ref_lookup by manifest.ref_lookup().add_node(node) too.  Also need to remove the node from disabled_lookup. Could either rebuild it or add a function to remove node. Probably simpler to rebuild it.